src/frontend/screens/PhotoPreviewModal/index.tsx

```diff
@@ -0,0 +1,497 @@
+            <Text style={{marginLeft: 4, color: WHITE, fontSize: 13}}>
+              {intl(m.headerDeleteButtonText)}
+            </Text>
+          </TouchableOpacity>
+        ) : null;
+      },
+    };
+  };
+}
+
+function CreatedByDeviceIdInfoItem({
+  projectId,
+  observationDocId,
+}: {
+  projectId: string;
+  observationDocId: string;
+}) {
+  const lang = useAppLanguageTag();
+
+  const {data: observation} = useSingleDocByDocId({
+    projectId,
+    docType: 'observation',
+    docId: observationDocId,
+    lang,
+  });
+
+  const {data: createdByDeviceId} = useDocumentCreatedBy({
+    projectId,
+    originalVersionId: observation.originalVersionId,
+  });
+
+  return (
+    <InfoItem
+      icon={
+        <MaterialIcons
+          name="devices"
+          size={20}
+          color={NEW_DARK_GREY}
+          allowFontScaling
+        />
+      }>
+      <BodyText selectable style={styles.primaryInfoText} numberOfLines={1}>
+        {
+          // TODO: Showing this ID is a temporary measure
+          // Ideally we use a format that is more human readable/friendly
+          createdByDeviceId.slice(0, 15)
+        }
+      </BodyText>
+    </InfoItem>
+  );
+}
+
+function getDeviceDetailsText({make, model}: {make?: string; model?: string}) {
+  const displayedParts: Array<string> = [];
+
+  if (make) {
+    displayedParts.push(make);
+  }
+
+  if (model) {
+    displayedParts.push(model);
+  }
+
+  if (displayedParts.length === 0) {
+    return null;
+  }
+
+  return displayedParts.join(' ');
+}
+
+function getCameraDetailsText(
+  {
+    fNumber,
+    orientation,
+  }: {
+    fNumber?: number;
+    orientation?: number;
+  },
+  formatMessage: IntlShape['formatMessage'],
+): string | null {
+  const displayedParts: Array<string> = [];
+
+  if (typeof orientation === 'number') {
+    try {
+      const layout = getPhotoLayout(orientation);
+
+      displayedParts.push(
+        layout === 'horizontal'
+          ? formatMessage(m.landscape)
+          : formatMessage(m.portrait),
+      );
+    } catch (err) {
+      captureException(err);
+
+      // TODO: What happens if layout cannot be determined or is missing?
+      // Should we still try to display something?
+    }
+  }
+
+  if (typeof fNumber === 'number') {
+    displayedParts.push(`𝒇 ${fNumber}`);
+  }
+
+  if (displayedParts.length === 0) {
+    return null;
+  }
+
+  return displayedParts.join(' — ');
+}
+
+function getPhotoDetailsText(
+  {
+    width,
+    height,
+    storageSize,
+  }: {
+    width: number;
+    height: number;
+    storageSize?: number;
+  },
+  formatMessage: IntlShape['formatMessage'],
+): string | null {
+  const displayedParts: Array<string> = [];
+
+  // TODO: Should this be translated?
+  displayedParts.push(`${width} x ${height}`);
+
+  if (typeof storageSize === 'number') {
+    displayedParts.push(
+      formatMessage(m.imageStorageSize, {
+        value: Math.max(bytesToMegabytes(storageSize), 0.01).toFixed(2),
```

Comment:
rounding to the hundredth made sense to me, with a minimum value of `0.01` being shown. can't imagine many images will actually be smaller than that but just in case.